tests/ui/wf/wf-trait-associated-type-region.stderr

```diff
@@ -2,10 +2,12 @@ error[E0309]: the associated type `<Self as SomeTrait<'a>>::Type1` may not live
   --> $DIR/wf-trait-associated-type-region.rs:9:18
    |
 LL |     type Type2 = &'a Self::Type1;
-   |                  ^^^^^^^^^^^^^^^
+   |                  ^^^^^^^^^^^^^^^ ...so that the reference type `&'a <Self as SomeTrait<'a>>::Type1` does not outlive the data it points at
    |
-   = help: consider adding an explicit lifetime bound `<Self as SomeTrait<'a>>::Type1: 'a`...
-   = note: ...so that the reference type `&'a <Self as SomeTrait<'a>>::Type1` does not outlive the data it points at
+help: consider adding an explicit lifetime bound...
```

Comment:
Maybe this should just be an inline suggestion, rather than a verbose one. The ordering between the `help`s here is kinda hard to read, unless you are expecting it to be two-part.

Reply:
added a new label as a premise to the following "so ..". It is more verbose though.
```
LL | trait SomeTrait<'a> {
   |                 -- the associated type `<Self as SomeTrait<'a>>::Type1` must be valid for the lifetime `'a` as defined here...
LL |     type Type1;
LL |     type Type2 = &'a Self::Type1;
   |                  ^^^^^^^^^^^^^^^ ...so that the reference type `&'a <Self as SomeTrait<'a>>::Type1` does not outlive the data it points at
````